### DISCUSSION TITLE
Speak Readings

### DISCUSSION THREAD

**Discussion by Atlas-Night-Out** [NightscoutFoundation/xDrip#2666] [General]
Speak readings carries on reading out the last reading even if it lost connections to your CGM giving you false readings if you are unable to fully check the screen!

**Comment by Navid200** [comment]
I have seen this request before.  I will find it.

If you set up missed reading alert, shouldn't you also hear that alert?

**Comment by Atlas-Night-Out** [comment]
I thought it might have been removed. No, I do not read all the alerts Im too busy doing other things to help people. I have too many notifications to notice them all. Why remark these things to me? I have a life other than reading notifications! If it was on Discord Id be able to look at my own pleasure and not have to look through lists of lists of notifications. Why _ think its a good idea to have xdrip on there? :-) _

**Reply by Navid200** [reply]
I don't follow all your remarks.

If something my life depends on is malfunctioning, I would like to know so that I can act and fix it.
That's why the missed reading alert exists.  By default, that alert is disabled.  The user can enable it or not.

When there are missed readings for a while, I believe the reading on the main screen has a strike through if I'm not mistaken.
I agree that spoken readings should reflect that under the same conditions .  May be it should read the reading but continue with "stale", or "missed readings".

Please let me investigate and confirm that this is not intentional.  If not, I will ask you to convert this to an issue so that we can track it and add it based on priority.  But, please wait and don't do it yet.

**Reply by Atlas-Night-Out** [reply]
Thanks @Navid200  yes I know about the aler but it should still stop reading out the readings no matter what! If I do not have a screen to see it due to rain and mobile is out of site and Im not sure if settings have changed due to other issues as in I might be at a meeting and need it to not alert me when it does I had that 2 weeks ago and whatever I did I could not stop the noise while a meeting was going on and it was very annoying but also disturb everybody in the room. So I m forever not wanting to do anything in case these alarms go off yet I need the warning to alert me and the vibrate does that fine during the days but at night is of no use. 

What I'm trying to say is yes it's great an alert to let me know something is wrong but that does not tell me what it is and for all I know it could be Im low! So I would prefer it that we don't have to be guessing on if low or in a hypo or what sound does what it once a voice prompt to say your disconnected or something.  

It's not done it with me for a while as I know to not fully trust it if my mobile is out of sight but I thought it's best to report it to make it aware to be safer on it or a way for it to be safer.

I'm not at all sure what I did on the other post now was a while back and I think it was my first attempt to post anything on here! I think I get emails for lost on here and sometimes I get lost on too much going on! So it's nice to have clues whats going on so I leave it to get all notifications on here just me not paying attention as I have not seen anything or missed where I posted it.  

The screen strike-through is of no use when your mobile in out of site on a Bike in the rain, and wind and trying to keep up with the rest of the people I'm with I do not want to have to keep stopping to get my mobile out to see a strikethrough. Fine if in a Supper market not fine on a bike or if you were skying or anything where you have thick gloves on and can take you 10 minutes to do something to just check! Yes, its great we at least have it :-) just not at that moment in time can kill you! 

It's a bit like forgetting to turn the volume up on your mobile to be alerted at night I don't do that a few times too but with luck, Id left the vibrate on and that was enough to alert me that time!

It's the voice prompts Id rather trust more than me needing a Garmin to tell me my readings and also a Garmins screen is also not on all of the time to save battery on it, as that also does not get a strike through on lots of the APPS for it I've noticed one comes up with a B.T on it, The one that does get a strike though is again hidden on the Garmin for others to be aware of its use, but I know to use it as its more trusted to me and I think that is a xdrip APP too. one whereas the others I think are more on Nightscout so I get no strike through on them.

I will post a pic of my alert setting on my lows but can you tell me why when I turned the sound down and turned things off got sound coming from my mobile I think was a software hitch (a one-off) but I do not fact it happening again like tonight Im suppose to go to the meeting but not going! :-) just in case it goes off again. Or until I'm sure I got it in a position I can trust it to let me know and not the rest of the people in the room :-) so embarring but not only that very disturbing to the person doing their speech! while I panic :-) 

Also if an alert I said to me I do not want that alert to keep going off in my ear for all ride or until I can stop to turn it off. Be great If I could but very hard with no hands! Even if I got voice prompts I would also not be wanting to hear it for an hour it would be just a case of it making sure I know about it so maybe a 5 mins one then it shut up after that! :-) Maybe we can call it (noBT)  for no Bluetooth connection? 
![460 alert](https://user-images.githubusercontent.com/83882497/217872373-cc0b9930-5dbd-4b51-8125-cb5fa3ce6703.jpg)

**Reply by Navid200** [reply]
You can use the Ascending Volume Profile: https://navid200.github.io/xDrip/docs/Ascending-volume-profile.html

That profile, by default, starts silent only vibrating.
After 3 minutes, if you have not snoozed it, it will also sound the alert at only 33% volume.  Then, on the following minute, it will increase the volume.  The reason this profile was created years ago was exactly the alert going off while you are in a meeting.

If you don't want the alert to make a sound at all and only vibrate, you can change the volume profile to "Vibrate only".
https://navid200.github.io/xDrip/docs/Volume-profiles.html

**Comment by Atlas-Night-Out** [comment]
Is there a way to separate these alarms to do different things for different times? E.g like keep this one as I like this for the night! And then me have another for the meeting. so then it's easy to swap over. And one for no hands-free :-)  I was also unable to see when the alerts went off in a dark place and my mobile in my pocket, which I can quickly snooze now by using the watch, if it works at the time it goes off. Different on  BIKE as you do not want to have to keep hearing the same thing when you know about it but cannot snooze it as thick gloves prevent you from it. Even the voice prompts are annoying after 10 mins of hearing them and you have eaten but waiting for it to catch up and go off.

Maybe an alexa or google button alert for a bike that waterproof to snooze it

**Reply by Navid200** [reply]
The Glucose level alerts have a time of day setting.
You can create an identical copy of your alert and call it "Low_Night".

You need to disable where you see in your screenshot saying "all day".  Then, it will let you edit the times.  You can have as many as you want.  So, you can have one for night, one for morning, one for afternoon and one for evening.

You can set up each differently to behave the way you desire. 

If for one of those periods, you only want it to vibrate, you can select a sound clip for it that is silent.  You can even download a silent sound clip for that if you don't have one.

**Reply by Atlas-Night-Out** [reply]
I did have a fav sound that I had from the start of using xdrip but silly Samsung restored my mobile to factory defaults and I had not transferred it over. I somehow have it on my old mobile but it's hidden deep to get a copy of it. Where are the files being kept now as Samsung keep shifting things around? I think it was a sound from their lists that they change each mobile version was called On Time though Im never on Time :-) what format do they need to be and where do they need to be placed? Can they go in the Download folder? I normally have a headset when on a bike that is not blocking my ears too and really comfy just not me not being able to do anything to shut it up as I always have food on me and can get out of the lows even with the gloves on very hard.
Thanks!

**Reply by Navid200** [reply]
I know mp3 works.  Other sound file formats may work too.  But, I am not sure.
You can place the sound file anywhere you like on your phone.  When you tap on choose file, on the edit Alert page, you need to tap on Custom Sound.  Then, you can find the file you have downloaded wherever it is and select it.  Yes, it can also be in the Download folder.

**Reply by Atlas-Night-Out** [reply]
Thanks will do a test run on it tonight!

**Reply by Atlas-Night-Out** [reply]
Will it still read out my readings if I add a silent sound file? Changed it to this just hope it works now 
![xdrip 2 alarms 460](https://user-images.githubusercontent.com/83882497/217907174-31f7d450-7435-4e79-a659-4d6ff4c5d568.jpg)

**Reply by Navid200** [reply]
The sound file, silent or not, affects the alert you choose it for only.   It has nothing to do with speak readings.

**Reply by Atlas-Night-Out** [reply]
The test run seemed to go great thank you! Now the on bike test when there is a fix for it can you let me know, please? Thanks!

**Comment by jamorham** [comment]
In the code for speaking readings the very first check we do is to check the age of the reading and if its more than 4 minutes old then we don't speak it. Based on this I'm confused how it is possible for it to speak old readings as mentioned here.

**Reply by Navid200** [reply]
Didn't know that.  Testing now.

**Reply by Navid200** [reply]
I set up xDrip as a Nightscout follower and enabled speak readings.  After confirming that it was speaking the readings, I disabled internet connectivity of the phone.
The speak reading feature never spoke again.
So, I confirm that, at least in the test I described, the problem cannot be recreated.

**Comment by Navid200** [comment]
@Atlas-Night-Out Would you please explain what version of xDrip you are using, and what your hardware data source is?

**Comment by Atlas-Night-Out** [comment]
I was confused in a coma it telling me my last readings and they where false. I'm not talking about now today as things like this I don't come around to them that often thank god! But when it as done it just keep says the last readings! that is about the speak readings out every 5 mins. So you might of changed things already? Very rare too now I lose connections from Dexcom which is also a great help as I then have less a chance a failed sensors at that moment in time Im in rain! The version date is 2023-01.16 If I disable Bluetooth I still get readings read out even tonight I turned it off and I still got it read out to me this is not wifi to the sensor its Bluetooth! When Im out its still Bluetooth as I have it direct to me to be safer. G5/G6/Dex1 Transmitter and its one of them bugs I cannot re produce I just know it can do it! been there twice and both times I was in a bad way and very confused as I was in the hypo unknowing. I never use Nightscout while out to be trusting it as Nightscout can easily lose connections. I also do not use a follower to do things for myself. If it says nothing Id still need to know something from xdrip to let me know! Its the letting me know that I need to know not me to have to think to know whats happening in my pockets on my mobile I cannot look! So time can fly and  I have eno clue if it was 5 mins or 30 mins when Im on a bike that's why I cycle to relax and get away from it all including diabetes! :-) if I can :-)

**Comment by Atlas-Night-Out** [comment]
Another thing as well when it does it its a failed sensor, so it might be that what's confusing the software, as I find out later the sensor has come out of me. So this maybe why we cannot re-make this issue to see if the reading keep reading out. I will see how it goes this year!

**Reply by Navid200** [reply]
I am not sure what happened.
This conversation so far has been about reading blood glucose values.  But, there is something else that I wonder may be causing a confusion.

By default, xDrip continues to alert you even if you have no more readings.
There is a setting that disables alerts if there are no readings.  It is called "Suppress alerts if missed readings".   I wonder if that's what you have experienced.
